code/modules/mob/living/carbon/life.dm

```diff
@@ -245,6 +275,33 @@
 			if(!.)
 				return FALSE //to differentiate between no internals and active, but empty internals
 
+// Make corpses rot, emitting miasma
+/mob/living/carbon/proc/rot()
+	// Properly stored corpses shouldn't create miasma
+	if(istype(loc, /obj/structure/closet/crate/coffin)|| istype(loc, /obj/structure/closet/body_bag) || istype(loc, /obj/structure/bodycontainer))
+		return
+
+	// No decay if formaldehyde in corpse
+	if(typesof(/datum/reagent/toxin/formaldehyde) in reagents.reagent_list)

```

Comment:
I'd suggest requiring a minimum amount of formaldehyde so you can't just sprinkle it with a dropper (since it never decays)

Reply:
Done, require 20u at least.